### DISCUSSION TITLE
Pi HAT compatibility

### DISCUSSION THREAD

**Discussion by okuuva** [gokrazy/gokrazy#141] [Q&A] [answered]
Has anyone tried running gokrazy with some Pi HATs? My main interest is 3G / LTE modems since I have a need for running Rasperry Pi without a readily available network and still would have to be able to connect to it remotely.

**Comment by stapelberg** [comment, marked as answer]
Sure, I have used USB and serial adapter HATs with gokrazy before.

As long as the device in question is supported by the upstream linux kernel, it usually can be made work. Maybe the kernel build configuration option for the driver needs to be enabled, though.

Check out https://gokrazy.org/development/non-go/ if you need software not written in Go. You can also run docker containers which makes it even easier to quickly test out software: https://gokrazy.org/packages/docker-containers/

**Reply by okuuva** [reply]
Ok, thank you for the info! I'll give it a go (pun not intended) once the modem I ordered arrives and report back either here or/and with a pull request to the docs :)

**Reply by stapelberg** [reply]
Sounds good!
